commands/util/binary.js

```diff
@@ -0,0 +1,187 @@
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("subtract")
+        .setDescription("Subtract two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+          )
+          .addIntegerOption(option =>
+            option
+              .setName("second")
+              .setDescription("The second number")
+              .setRequired(true)
+        )
+    )
+    .addSubcommand(subcommand =>
+      subcommand
+        .setName("multiply")
+        .setDescription("Multiply two binary numbers")
+        .addIntegerOption(option =>
+          option
+            .setName("first")
+            .setDescription("The first number")
+            .setRequired(true)
+        )
+        .addIntegerOption(option =>
+          option
+            .setName("second")
+            .setDescription("The second number")
+            .setRequired(true)
+        )
+    ),
+
+  async execute(interaction) {
+    try {
+      const subcommand = interaction.options.getSubcommand();
+
+      if (subcommand === "encode") {
+        try {
+          const integer = interaction.options.getInteger("number");
+          const binary = integer.toString(2);
+
+          const embed = new EmbedBuilder()
+            .addFields(
+              { name: 'Number', value: `**${integer}**`, inline: true },
+              { name: 'Binary', value: `**${binary}**`, inline: true },
+            );
+
+          await interaction.reply({ embeds: [embed] });
+        } catch (error) {
+          await interaction.reply({ content: `An error occurred while encoding: ${error.message}`, ephemeral: true });
+        }
+
+      } else if (subcommand === "decode") {
+        try {
+          const binary = interaction.options.getInteger("binary");
+          if (!/^[01]+$/.test(binary.toString())) {
+            throw new Error("Input must be a valid binary number.");
+          }
+          const integer = parseInt(binary, 2);
+
+          const embed = new EmbedBuilder()
+            .addFields(
+              { name: 'Binary', value: `**${binary}**`, inline: true },
+              { name: 'Number', value: `**${integer}**`, inline: true },
+            );
+
+          await interaction.reply({ embeds: [embed] });
+        } catch (error) {
+          await interaction.reply({ content: `An error occurred while decoding: ${error.message}`, ephemeral: true });
+        }
+
+      } else if (subcommand === "add") {
+        try {
+          const first = interaction.options.getInteger("first");
+          const second = interaction.options.getInteger("second");
+          if (!/^[01]+$/.test(first.toString()) || !/^[01]+$/.test(second.toString())) {
```

Comment:
Not a problem right now but later on down the line, please separate the boolean and make it its own variable for better code readability